File: src/main/java/no/skatteetaten/aurora/databasehotel/service/oracle/OracleJdbcUrlBuilder.java
package no.skatteetaten.aurora.databasehotel.service.oracle;

import no.skatteetaten.aurora.databasehotel.service.JdbcUrlBuilder;

public class OracleJdbcUrlBuilder implements JdbcUrlBuilder {
    private final String service;

    public OracleJdbcUrlBuilder(String service) {
        this.service = service;
    }

    @Override
    public String create(String dbHost, int port) {

        return String.format("jdbc:oracle:thin:@%s:%d/%s", dbHost, port, service);
    }
}

File: src/main/kotlin/no/skatteetaten/aurora/databasehotel/service/oracle/OracleJdbcUrlBuilder.kt
package no.skatteetaten.aurora.databasehotel.service.oracle

import no.skatteetaten.aurora.databasehotel.service.JdbcUrlBuilder

class OracleJdbcUrlBuilder(private val service: String) : JdbcUrlBuilder {

    override fun create(dbHost: String, port: Int, database: String?): String =
        "jdbc:oracle:thin:@$dbHost:$port/$service"
}
